contracts/src/v0.8/dev/keeper2_0/interfaces/KeeperRegistryInterface2_0.sol

```diff
@@ -64,6 +64,28 @@ struct State {
   bool paused;
 }
 
+/**
+ * @notice relevant state of an upkeep which is used in transmit function
+ * @member executeGas the gas limit of upkeep execution
+ * @member maxValidBlocknumber until which block this upkeep is valid
+ * @member paused if this upkeep has been paused
+ * @member target the contract which needs to be serviced
+ * @member amountSpent the amount this upkeep has spent
+ * @member balance the balance of this upkeep
+ * @member lastPerformBlockNumber the last block number when this upkeep was performed
+ */
+struct Upkeep {
```

Comment:
not fully sure if this makes sense in the interface. It's more of an internal detail of registry on how it stores the upkeep data.

The whole upkeep struct exposed to users in interface is UpkeepInfo

Reply:
as i mentioned in the comment below, we cannot export a struct defined within a contract.
that leaves us with either moving Upkeep struct to this interface or defining a same struct in transcoder.
i think this is better bc everything is already public on chain including how Upkeep is defined even we keep Upkeep struct on the registry. i feel we are not hiding it from users like in web2 world?